File: backend/etablissement-service/src/main/java/fsm/miaad/models/Filiere.java
package fsm.miaad.models;

import lombok.*;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Builder
public class Filiere {
    private Long id;
    private String nom;
    private int nombreAnnees;
    private int nombreSemestresParAnnee;
//    private StructureEnseignement structure;
    private ResponsableFiliere responsable;
    private List<Long> modulesId;
    private List<Semestre> semestres;

    



}

File: frontend/admin-global/src/app/models/etablissementModel/FiliereModel/filiere.ts
import { Responsable } from "./responsable"
import { Semestre } from "./semestre"

export class Filiere {
    id!: number
    nom!: string
    nombreAnnees!: number
    nombreSemestresParAnnee!: number
    responsable!: Responsable
    modulesId!: number[]
    semestres!: Semestre[]
}